### ISSUE TITLE
Support more methods of installing plugins

### ISSUE THREAD

**Issue by LukeTowers** [wintercms/winter#77]
We should support more ways to install plugins:

- Upload ZIP
- Specify git repo (only if Git binary supported on the system)
- Specify composer package name
- (only on plugin:install command) Specify local folder

These methods should exist in the plugin:install command with a `--from=` option and in the backend plugin manager.

**Comment by mjauvin** [comment]
We should take this opportunity to unify both the UI and console installer/update code paths

**Comment by tschallacka** [comment]
I would also add as requirement that a specific version can be specified of a plugin, when not uploading as zip. People might end up running on an outdated software version because of reasons, where only a specific version is compatible with, or they want a feature that only an older version offered in a way they like it(think ui changes etc...)

**Comment by LukeTowers** [comment]
@tschallacka I don't think we want to encourage people to use old versions of plugins, if they absolutely cannot have the updated version they can always just install it manually via composer or putting the files directly into the project.
